File: app-android/src/main/java/com/ustadmobile/port/android/netwokmanager/EmbeddedHttpdService.java
package com.ustadmobile.port.android.netwokmanager;

import android.app.Service;
import android.content.Intent;
import android.os.Binder;
import android.os.IBinder;
import android.support.annotation.Nullable;

import com.ustadmobile.core.impl.UMLog;
import com.ustadmobile.core.impl.UstadMobileSystemImpl;
import com.ustadmobile.port.android.impl.http.AndroidAssetsHandler;
import com.ustadmobile.port.sharedse.impl.http.EmbeddedHTTPD;

import java.io.IOException;

public class EmbeddedHttpdService extends Service {

    private EmbeddedHTTPD httpd;

    private LocalServiceBinder mBinder = new LocalServiceBinder();

    public static final String ANDROID_ASSETS_PATH = "/android-assets/";

    public class LocalServiceBinder extends Binder {

        public EmbeddedHTTPD getHttpd(){
            return httpd;
        }

    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        return START_STICKY;
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return mBinder;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        httpd = new EmbeddedHTTPD(0, getApplicationContext());
        httpd.addRoute(ANDROID_ASSETS_PATH +"(.)+",  AndroidAssetsHandler.class,
                getApplicationContext());
        try {
            httpd.start();
        }catch(IOException e) {
            UstadMobileSystemImpl.l(UMLog.CRITICAL, 0, "Could not start httpd server");
            throw new RuntimeException("Could not start httpd server", e);
        }
    }

    @Override
    public void onDestroy() {
        httpd.stop();
        super.onDestroy();
    }
}

File: sharedse/src/androidMain/kotlin/com/ustadmobile/sharedse/network/EmbeddedHttpdService.kt
package com.ustadmobile.sharedse.network

import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import com.ustadmobile.port.sharedse.impl.http.EmbeddedHTTPD
import org.kodein.di.DIAware
import org.kodein.di.android.di
import org.kodein.di.instance

@Deprecated("Use httpd from dependency injection instead")
class EmbeddedHttpdService : Service(), DIAware {

    override val di by di()

    private val mBinder = LocalServiceBinder()

    val httpd: EmbeddedHTTPD by instance()

    inner class LocalServiceBinder : Binder() {

        fun getHttpd(): EmbeddedHTTPD {
            return httpd
        }
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        return Service.START_STICKY
    }

    override fun onBind(intent: Intent): IBinder? {
        return mBinder
    }

    override fun onCreate() {
        super.onCreate()

//        val appDatabase = UmAccountManager.getActiveDatabase(applicationContext)
//        val appRepository = UmAccountManager.getRepositoryForActiveAccount(applicationContext)
//        httpd = EmbeddedHTTPD(0, applicationContext, appDatabase = appDatabase,
//                repository =  appRepository)
//        httpd.addRoute("$ANDROID_ASSETS_PATH(.)+", AndroidAssetsHandler::class.java,
//                applicationContext)
//        httpd.UmAppDatabase_AddUriMapping(UmAccountManager.getActiveDatabase(this),
//                Gson(), "/test/", false, "/rest/UmAppDatabase")
//        try {
//            httpd.start()
//            UMLog.l(UMLog.INFO, 0, "Started embedded HTTP server on port: ${httpd.listeningPort}")
//        } catch (e: IOException) {
//            UMLog.l(UMLog.CRITICAL, 0, "Could not start httpd server")
//            throw RuntimeException("Could not start httpd server", e)
//        }

    }

    override fun onDestroy() {
        httpd.stop()
        super.onDestroy()
    }

    companion object {

        val ANDROID_ASSETS_PATH = "/android-assets/"
    }
}
